zipline/protocol.py

```diff
@@ -153,6 +216,12 @@ def __init__(self):
         self.start_date = None
         self.positions_value = 0.0
 
+        self.benchmark_asset = benchmark_asset
+
+        self._data_portal = data_portal
+        self._current_dt_callback = current_dt_callback
+        self._expiring_cache = ExpiringCache()
```

Comment:
I still need to solve the problem of hiding these private attributes (or at least the data portal)